### DISCUSSION TITLE
Problem trying to replicate example simulations

### DISCUSSION THREAD

**Discussion by martoconnh** [Parcels-code/plasticparcels#28] [General]
Hi there!

I am just starting with PlasticParcels, so I wanted to try by replicating the [Italy example](https://plastic.oceanparcels.org/en/latest/examples/example_Italy_coast.html) on your website. But I am getting the following error when doing  `fieldset = pp.constructors.create_fieldset(settings)` :

```
Traceback (most recent call last):
  File "c:\Users\martino.rial\Desktop\scripts\test_plasticparcels.py", line 50, in <module>
    fieldset = pp.constructors.create_fieldset(settings)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\martino.rial\AppData\Local\Programs\Python\Python312\Lib\site-packages\plasticparcels\constructors.py", line 104, in create_fieldset
    fieldset = create_hydrodynamic_fieldset(settings)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\martino.rial\AppData\Local\Programs\Python\Python312\Lib\site-packages\plasticparcels\constructors.py", line 43, in create_hydrodynamic_fieldset
    filenames = {'U': {'lon': ocean_mesh, 'lat': ocean_mesh, 'depth': wfiles[0], 'data': ufiles},
                                                                      ~~~~~~^^^
IndexError: list index out of range
```

I'm still not really sure about how everything works, but may this be happening because I have to download the hydrodynamic data that are declared on the settings.json file for this particular case?

```
"ocean": {
        "modelname": "NEMO0083",
        "directory": "/storage/shared/oceanparcels/input_data/MOi/",
        "filename_style": "psy4v3r1/psy4v3r1-daily_",
        "ocean_mesh": "domain_ORCA0083-N006/PSY4V3R1_mesh_hgr.nc",
...
}
```

I don't really think this might be the problem, as I would have got an error indicating that these input files could not be found in my device. But I figured I'd ask anyway. Are these databases intrinsic of OceanParcels or something like that?

Thank you !! (And sorry if this is a beginner mistake)

**Comment by michaeldenes** [comment]
Hi @martoconnh, thank you for your interest in using `PlasticParcels`! I'm still in the process of updating all the documentation (you'll see over the coming weeks there will be quite a few new commits to the main branch of this project).

You are correct, you will need to download the hydrodynamic model data, biogeochemical model data, wind data, and wave data. We've developed `PlasticParcels` to be used with the [NEMO Global 1/12° Analysis & Forecast psy4v3r1-daily (GLO12)](https://www.mercator-ocean.fr/en/solutions-expertise/accessing-digital-data/product-details/?offer=4217979b-2662-329a-907c-602fdc69c3a3&system=d35404e4-40d3-59d6-3608-581c9495d86a) hydrodynamic model. The associated biogeochemistry model is the [PISCES-NEMO3.6 Global 1/4° Analysis & Forecast biomer4v2r1-weekly (BIO4)](https://www.mercator-ocean.fr/en/solutions-expertise/accessing-digital-data/product-details/?offer=8d0c01f3-81c7-0a59-0d06-602fdf63c5b6&system=dc40b324-7de7-0732-880b-5d9dcf7d344a) model. And for wind and waves, we use the ERA5 Reanalysis products via the CDSAPI (Climate Data Store API).

I'll be writing more documentation soon on where to download this data, and how it is used by `PlasticParcels`.

You can of course (with slight modifications where necessary) use your own hydrodynamic model data, along with a biogeochemistry model data (required for the biofouling behaviour), and wind and wave data (required for wind-induced drift, and Stokes drift behaviours, respectively). Expect more documentation soon to help explain all this, but in the meantime please let me know if I can help!

Cheers,
Michael

**Comment by suraj8070** [comment]
Hi,

I was also trying to test the Italy example to familiarise myself with plasticparcels even though I intend to try over another regional domain. How about using [Hydrodynamic_data](https://data.marine.copernicus.eu/product/GLOBAL_MULTIYEAR_PHY_001_030/description) and ERA5 wind and waves to begin with, as I am not interested in having biofouling as of now. It would be great if you could help with the readme regarding the preparation of the input data (temporal frequency, spatial resolution of era5 v/s hydrodynamic, etc.).

Thanks.
Suraj

**Reply by michaeldenes** [reply]
Hi @suraj8070, if you already have a regional dataset in mind, you can choose to setup the fieldset (with the hydrodynamic, wind, and wave data) simply using `parcels` (https://github.com/OceanParcels/Parcels), you don't necessarily need the specific GLO12 and BIO4 models we used for the examples.

To answer your question on spatiotemporal resolution of the datasets, they don't need to be aligned either. You can use daily-averaged velocity data with hourly wind data, on different spatial grids (e.g. a 1/12th degree C grid for one set of data, and a 1/4th degree A grid for another), that's no problem!

**Comment by s0fiateixeira** [comment]
Hi everyone!

I’m currently trying to run the Italy coast example using hydrodynamic data from [Copernicus](https://data.marine.copernicus.eu/product/GLOBAL_MULTIYEAR_PHY_001_030/files?path=GLOBAL_MULTIYEAR_PHY_001_030%2Fcmems_mod_glo_phy_my_0.083deg_P1D-m_202311%2F2020%2F01%2F&subdataset=cmems_mod_glo_phy_my_0.083deg_P1D-m_202311) for January 2020 and waves + wind data from the links provided on the [PlasticParcels documentation page](https://plastic.parcels-code.org/en/latest/index.html#required-data) for the same data frame.

After some debugging, I noticed that the example seemed to require biogeochemical data (which I don’t intend to use) as well as vertical velocity W values, which I could not find anywhere. To work around this, I created some dummy values.

I also received warnings about mismatched latitude orientation, so I attempted to fix this by manually adjusting the latitudes for waves_latfixed.nc and wind_latfixed.nc before using them on the tutorial.

However, when I run the example, the results look incorrect. The trajectories appear as straight lines, and the particles in the second plot appear on land and are very dispersed. It seems like the FieldSet may not be reading the data properly, but I haven’t been able to identify where the issue is. I’m sorry if this is a beginner mistake. I just wanted to run the Italy coast example to understand the workflow using OceanParcels / PlasticParcels, but I’m struggling to see how to fix this.

I will leave here my code, terminal output, the JSON settings files and the resulting final plot.


[tutporial.py](https://github.com/user-attachments/files/23346549/tutporial.py)

[tutporial2.txt](https://github.com/user-attachments/files/23346567/tutporial2.txt)

[Italy_coast_settings.json](https://github.com/user-attachments/files/23346571/Italy_coast_settings.json)

<img width="2401" height="881" alt="italy_debug_results" src="https://github.com/user-attachments/assets/4e1f7bfb-8760-432f-9702-3544baec5452" />


Thank you in advance to anyone who takes the time to help!
Sofia

**Reply by michaeldenes** [reply]
Hi @s0fiateixeira, sorry for the late response! I've been (and am) quite busy at the moment...
I've had a quick look at your python script, and I think all that your missing is setting the units of the Stokes and Wind fields. The `(Stokes_U, Stokes_V)` and `(Wind_U, Wind_V)` fields are loaded in with units of m/s, but `parcels` (and hence `plasticparcels`) performs all of the advection in units of deg/s. So you'll need to add the following lines:

```python
from parcels.tools.converters import Geographic, GeographicPolar

# Add this to roughly line 228 - between creating stokes_fs, and adding each field to the fieldset
stokes_fs.Stokes_U.units = GeographicPolar()
stokes_fs.Stokes_V.units = Geographic()

# Add this to roughly line 273, between creating wind_fs, and adding each field to the fieldset
wind_fs.Wind_U.units = GeographicPolar()
wind_fs.Wind_V.units = Geographic()
```

Let me know if that runs, and apologies again for the delay! Good luck!

**Reply by s0fiateixeira** [reply]
Thanks a lot for your help. That completely solved the issue!
Apologies for the late response on my side. I really appreciate you taking the time to help.

<img width="2401" height="881" alt="italy_debug_results" src="https://github.com/user-attachments/assets/b7565fef-bc2d-4247-bb5c-d6c9b802a67c" />
